library/manageiq_custom_attributes.py

```diff
@@ -0,0 +1,254 @@
+---
+module: manageiq_custom_attributes
+short_description: add, update, delete an entity custom attributes in ManageIQ
+requirements: [ ManageIQ/manageiq-api-client-python ]
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  entity_name:
+    description:
+      - the entity name in manageiq to which the custom attributes belongs
+    required: true
+    default: null
+  entity_type:
+    description:
+      - the entity type in manageiq to which the custom attributes belongs
+    required: true
+    default: null
+  state:
+    description:
+      - the state of the custom attributes
+      - On present, it will add the custom attributes to the entity if not
+      already exist, or update the custom attributes if the associated data
+      is different
+      - On absent, it will delete the custom attributes from the entity if exist
+    required: false
+    choices: ['present', 'absent']
+    default: 'present'
+  custom_attributes:
+    description:
+      - the custom attributes of the entity
+    required: true
+    default: null
+'''
+
+EXAMPLES = '''
+# Add custom attributes for Openshift Containers Provider in ManageIQ
+  manageiq_custom_attributes:
+    entity_type: 'provider'
+    entity_name: 'openshift01'
+    state: 'present'
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+    custom_attributes: '[
+        {"name": "ca1", "value": "value 1"},
+        {"name": "ca2", "value": "value 2"}
+    ]'
+'''
+
+
+class ManageIQ(object):
+    """ ManageIQ object to execute various operations in manageiq
+
+    url      - manageiq environment url
+    user     - the username in manageiq
+    password - the user password in manageiq
+    """
+
+    supported_entities = {'vm': 'vms', 'provider': 'providers'}
+
+    def __init__(self, module, url, user, password):
+        self.module        = module
+        self.api_url       = url + '/api'
+        self.user          = user
+        self.password      = password
+        self.client        = MiqApi(self.api_url, (self.user, self.password))
+        self.changed       = False
+
+    def find_entity_by_name(self, entity_type, entity_name):
+        """ Searches the entity name in ManageIQ.
+
+            Returns:
+                the entity id if it exists in manageiq, None otherwise.
+        """
+        entities_list = getattr(self.client.collections, ManageIQ.supported_entities[entity_type])
+        return next((e.id for e in entities_list if e.name == entity_name), None)
+
+    def get_entity_cas(self, entity_type, entity_id):
```

Comment:
I agree, will change to `get_entity_custom_attributes` since you can also get all custom attributes but this method only handles the ones set to the entity.